packages/modelviewer.dev/examples/scene-graph.html

```diff
@@ -105,11 +105,162 @@ <h2 class="demo-title">Change Material Base Color</h2>
   </div>
 
 
+  <div class="sample">
+    <div id="demo-container-2" class="demo"></div>
+    <div class="content">
+      <div class="wrapper">
+        <div class="heading">
+          <h2 class="demo-title">Change Material Metalness and Roughness Factors</h2>
+        </div>
+        <example-snippet stamp-to="demo-container-2" highlight-as="html">
+          <template>
+
+<model-viewer id="sphere" camera-controls interaction-prompt="none" src="../shared-assets/models/reflective-sphere.gltf" alt="A 3D model of a sphere">
+  <div id="controls">
+    <input id="metalness" type="range" min="0" max="1" step="0.01" value="1" />
+    <input id="roughness" type="range" min="0" max="1" step="0.01" value="0" />
+  </div>
+</model-viewer>
+<script>
+const modelViewerParameters = document.querySelector("model-viewer#sphere");
+
+document.querySelector('#metalness').addEventListener('input', (event) => {
+  const [material] = modelViewerParameters.model.materials;
+  console.log(event.target.value);
+  material.pbrMetallicRoughness.setMetallicFactor(event.target.value);
+});
+
+document.querySelector('#roughness').addEventListener('input', (event) => {
+  const [material] = modelViewerParameters.model.materials;
+  material.pbrMetallicRoughness.setRoughnessFactor(event.target.value);
+});
+</script>
+          </template>
+        </example-snippet>
+      </div>
+    </div>
+  </div>
+
+  <div class="sample">
+    <div id="demo-container-3" class="demo"></div>
+    <div class="content">
+      <div class="wrapper">
+        <div class="heading">
+          <h2 class="demo-title">Swap textures - Diffuse/Normal/MetallicRoughness</h2>
+        </div>
+        <example-snippet stamp-to="demo-container-3" highlight-as="html">
+          <template>
+
+<model-viewer id="lantern" camera-controls interaction-prompt="none" src="../shared-assets/models/glTF-Sample-Models/2.0/Lantern/glTF-Binary/Lantern.glb" alt="A 3D model of a lantern">
+  <div id="controls">
+    <div>
+      <p>Diffuse</p>
+      <input id="diffuse" type="file" accept="image/png, image/jpeg"/>
```

Comment:
I'm open to push-back here, but I think this might be a clearer example if people don't have to bring their own textures. Could we just grab two or three interesting textures and make a drop-down to select between them for this and the following example?

Reply:
I've been thinking about this: giving the users a dropdown to choose from may be misinterpreted as a simple model swap under the hood...but yeah, the code should be enough to judge that it's not the case.